src/clib/pio_darray.c

```diff
@@ -162,6 +170,7 @@ int PIOc_write_darray_multi(const int ncid, const int *vid, const int ioid,
         }
     }
 
+    /* ??? */
```

Comment:
I have separately shared a set of slides with you that illustrate the two different rearrangement options provided.   The box rearranger will always have data (it could be fill data) to fill the entire array - that is the aggregate start and count values will completely describe one unlimited dimension unit of the array.  For the subset method this is not necessarily the case, areas of missing data may never be written.  In order to make sure that these areas are given the missing value a 'holegrid' is used to describe the missing points.   This is generally faster than the netcdf method of filling the entire array with missing values before overwriting those values later.